UefiCpuPkg/PiSmmCpuDxeSmm/PiSmmCpuStandaloneMm.c

```diff
@@ -9,6 +9,61 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 
 #include "PiSmmCpuCommon.h"
 
+//
+// RemainingTasks Done flag
+//
+BOOLEAN  mRemainingTasksDone = FALSE;
+
+/**
+  Perform the remaining tasks.
+
+**/
+VOID
+PerformRemainingTasks (
+  VOID
+  )
+{
+  EDKII_PI_SMM_MEMORY_ATTRIBUTES_TABLE  *MemoryAttributesTable;
+
+  if (!mRemainingTasksDone) {
+    PERF_FUNCTION_BEGIN ();
+
+    //
+    // gEdkiiPiSmmMemoryAttributesTableGuid should have been published after SmmCore dispatched all MM drivers (MmDriverDispatchHandler).
+    //
+    SmmGetSystemConfigurationTable (&gEdkiiPiSmmMemoryAttributesTableGuid, (VOID **)&MemoryAttributesTable);
+
+    //
+    // Set critical region attribute in page table according to the MemoryAttributesTable
+    //
+    if (MemoryAttributesTable != NULL) {

```

Comment:
When MemoryAttributesTable is NULL? Can we assert that the table is NOT NULL?

Reply:
as we talked offline,   MemoryAttributesTable depends on below bit check, it's not always  intsalled: EFI_MEMORY_ATTRIBUTES_RUNTIME_MEMORY_PROTECTION_NON_EXECUTABLE_PE_DATA.
i will refine the comment here.